hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -100,16 +103,11 @@ public MySQLDialect() {
 	}
 
 	public MySQLDialect(DatabaseVersion version) {
-		// Let's be conservative and assume people use a 4 byte character set
-		this( version, 4 );
+		this(version, null);
 	}
 
-	public MySQLDialect(DatabaseVersion version, int characterSetBytesPerCharacter) {
```

Comment:
IMO we should retain this constructor to allow subclasses provide a value to avoid access to the database on boot. The invocation order due to subclassing seems to become a problem here though. How about expecting subclass constructors to invoke some `finishInitialization` instead to avoid some of the potential issues?

Reply:
Well took that approach first, and I didn't like it. Neither approach is quite perfect but I settled on this design as the cleanest option.